wt-status.c

```diff
@@ -1007,6 +1007,8 @@ size_t wt_status_locate_end(const char *s, size_t len)
 	struct strbuf pattern = STRBUF_INIT;
 
 	strbuf_addf(&pattern, "\n%c %s", comment_line_char, cut_line);
+	strbuf_trim_trailing_newline(&pattern);
```

Comment:
That is unfortunately not quite enough, as the line could continue with additional text before the newline, in which case it would _not_ be a "scissors cut" line.

I wonder whether we can safely convert from CR/LF to LF before matching, or whether we have to invent a function to match complete lines.

Alternatively, since we know that we're looking for either CR/LF or LF, we could remove the trailing newline of `cut_line` (which is used in `wt_status_append_cut_line()` where we would have to add it explicitly), and then verify that the `cut_line` is followed by a newline.

Something like this:

```c
static inline starts_with_newline(const char *p)
{
    return *p == '\n' || (*p == '\r' && p[1] == '\n';
}

[...]

    if (starts_with(s, pattern.buf + 1) && starts_with_newline(s + pattern.len - 1))
        len = 0;
    else if ((p = strstr(s, pattern.buf)) && starts_with_newline(p + pattern.len))
        len = p - s + 1;
```

In any case, we will want to have a regression test verifying not only that this is broken without this patch and fixed with this patch, but also that it won't regress again in the future. The best home for this is probably `t/t7502-commit-porcelain.sh`.
```

Reply:
> I wonder whether we can safely convert from CR/LF to LF before matching, or whether we have to invent a function to match complete lines.

I prefer your idea about removing the newline from `cut_line[]` and testing for either line ending, though probably because I can understand the example despite having never worked in C before.

> In any case, we will want to have a regression test verifying not only that this is broken without this patch and fixed with this patch, but also that it won't regress again in the future. The best home for this is probably `t/t7502-commit-porcelain.sh`.

Regression tests seemed to be a better match for my limited skill-set, so I had a root-around in `t7502-commit-porcelain.sh` and am now somewhat discouraged. The test `git commit --cleanup=scissors -e -F text -a` [here](https://github.com/git-for-windows/git/blob/a98b073ae2b993892641bb1eb6ce707cc673d9ad/t/t7502-commit-porcelain.sh#L255) will *always* pass because no interactive editor is opened and the user gets no chance to save the commit message with CRLF endings. 

Manufacturing a `text` file with CRLF endings for the above test seems fruitless, as CRLF line endings *appear* to be stripped by `git commit`.

Reply:
Maybe a "custom editor" might help? Something along these lines:

```sh
write_script lf-to-crlf.sh <<-\EOF
sed "s/\$/Q/" <"$1" | tr Q "\\015" >"$1".new &&
mv -f "$1".new "$1"
EOF
test_config core.editor "$PWD"/lf-to-crlf.sh
```